L1Trigger/Phase2DAQProducer/plugins/Phase2DAQProducer.cc

```diff
@@ -0,0 +1,218 @@
+#include "Geometry/CommonTopologies/interface/PixelGeomDetUnit.h"
+#include "Geometry/CommonTopologies/interface/PixelTopology.h"
+#include "Phase2RawToDigi/Phase2DAQProducer/interface/DTCAssembly.h"
+#include <unordered_map>
+
+#include "Phase2RawToDigi/Phase2DAQProducer/interface/Cluster.h"
+
+class Phase2DAQProducer : public edm::one::EDProducer<> {
+public:
+    explicit Phase2DAQProducer(const edm::ParameterSet&);
+    ~Phase2DAQProducer() override;
+
+private:
+    void produce(edm::Event&, const edm::EventSetup&) override;
+    
+    edm::EDGetTokenT<Phase2TrackerCluster1DCollectionNew> ClusterCollectionToken_;
+    const edm::ESGetToken<TrackerDetToDTCELinkCablingMap, TrackerDetToDTCELinkCablingMapRcd> cablingMapToken_;
+    const edm::ESGetToken<TrackerGeometry, TrackerDigiGeometryRecord> trackerGeometryToken_;
+    const edm::ESGetToken<TrackerTopology, TrackerTopologyRcd> trackerTopologyToken_;
+    
+    // const unsigned int Positive_Modules[108];
+    // const unsigned int Negative_Modules[108];
+
+    uint32_t assignNumber(const uint32_t& N);
+    void processClusters(TrackerGeometry::ModuleType moduleType,
+                                           const Phase2TrackerCluster1DCollectionNew::DetSet& DetectorClusterCollection,
+                                           unsigned int dtc_id, unsigned int slink_id, unsigned int slink_id_within,
+                                           DTCAssembly& dtcAssembly);
+};
+
+Phase2DAQProducer::Phase2DAQProducer(const edm::ParameterSet& iConfig)
+    : ClusterCollectionToken_(consumes<Phase2TrackerCluster1DCollectionNew>(iConfig.getParameter<edm::InputTag>("Phase2Clusters"))),
+      cablingMapToken_(esConsumes()),
+      trackerGeometryToken_(esConsumes<TrackerGeometry, TrackerDigiGeometryRecord>()),
+      trackerTopologyToken_(esConsumes<TrackerTopology, TrackerTopologyRcd>())
+    //   Positive_Modules{30, 42, 54, 66, 78, 90, 102, 6, 18, /*...*/},
+    //   Negative_Modules{138, 150, 162, 174, 186, 198, 210, 114, /*...*/}
+{
+    produces<FEDRawDataCollection>();
+}
+
+Phase2DAQProducer::~Phase2DAQProducer() 
+{
+
+}
+
+void Phase2DAQProducer::produce(edm::Event& iEvent, const edm::EventSetup& iSetup) 
+{
+    // Retrieve TrackerGeometry and TrackerTopology from EventSetup
+    const TrackerGeometry& trackerGeometry = iSetup.getData(trackerGeometryToken_);
+    // const TrackerTopology& trackerTopology = iSetup.getData(trackerTopologyToken_);
+    
+    // Retrieve the CablingMap
+    const auto& cablingMap = iSetup.getData(cablingMapToken_);
+
+    // get EventID and RunID
+    unsigned int EventID = iEvent.id().event();
+
+    DTCAssembly dtcAssembly (EventID);
+
+    // Create FEDRawDataCollection to store the output
+    auto fedRawDataCollection = std::make_unique<FEDRawDataCollection>();
+
+    // Iterate through Cluster Collection
+    for (const auto& DetectorClusterCollection : iEvent.get(ClusterCollectionToken_)) 
```

Comment:
CMSSW conventions are that all class names or typedefs must start with a capital letter. But all variables and constants must start with a small letter. (This helps distinguish the two). So DetectorClusterCollection should start with small letter. Similar comments apply elsewhere in your code.

Reply:
done